packages/blade/src/components/Input/BaseInput/types.ts

```diff
@@ -10,6 +10,7 @@ export type StyledBaseInputProps = {
   handleOnBlur?: FormInputHandleOnEvent;
   handleOnKeyDown?: FormInputHandleOnKeyDownEvent;
   handleOnInput?: FormInputHandleOnEvent;
+  onClick?: MouseEventHandler<HTMLInputElement>;
```

Comment:
are we sending the entire click event object to consumers? for other even handlers like onChange, onFocus we pass only the selected args like name, value, etc.

another thing to notice here - the reason you might not be able to type this and you have to use `@ts-expect-error` is because the event handler is diff on native and web. what we can do is we can handle this as we have done in Button

```ts
onClick?: Platform.Select<{
    native: (event: GestureResponderEvent) => void;
    web: (event: React.MouseEvent<HTMLButtonElement>) => void;
  }>;
```

or we can also just pick this from the types of button

Reply:
> are we sending the entire click event object to consumers? for other even handlers like onChange, onFocus we pass only the selected args like name, value, etc.

I just realised I don't need any other value from event property in Dropdown (earlier while building Select I was not sure so I was using entire event object) so I changed it to be similar to onFocus and onBlur events where we only pass name and value.

This also fixed TS issues since I just used the existing type helpers from forms